server/courselist.rb

```diff
@@ -627,6 +627,22 @@ def run_url_parse_json(new_url)
       logger.debug "#{__method__}: #{__LINE__}: #{new_url}: request_body_ruby: +++#{request_body_ruby.inspect}+++"
       request_body_ruby
     end
+
+    # Get the canvas courses from the list provided by mpathways.
+
+    def getCanvasCourseList(course_data, userid)
+      # could the regexp be a constant?
```

Comment:
Could it?  Where would it be defined, in this method or elsewhere?  Would the constant hold the Regexp object?  (Is that even possible?)  Or would the constant hold the regex string?  If you're worried about the time it takes to compile a regex, maybe this instantiation should go somewhere else.  This method could reuse the object instead of recreating it each time the method's invoked.